src/transformers/hyperparameter_search.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from .integrations import (
+    is_optuna_available,
+    is_ray_available,
+    is_sigopt_available,
+    is_wandb_available,
+    run_hp_search_optuna,
+    run_hp_search_ray,
+    run_hp_search_sigopt,
+    run_hp_search_wandb,
+)
+from .trainer_utils import (
+    HPSearchBackend,
+    default_hp_space_optuna,
+    default_hp_space_ray,
+    default_hp_space_sigopt,
+    default_hp_space_wandb,
+)
+
+
+class HyperParamSearchBackendBase(ABC):
+    @abstractmethod
+    def name(self) -> str:
+        raise NotImplementedError
```

Comment:
Why is this method? It's just a class attribute.

Reply:
It was for the enforcement offered by `abc`.